### DISCUSSION TITLE
Website Development

### DISCUSSION THREAD

**Discussion by qreeves** [redeclipse/base#1153] [General]
As we move our essential website services to GitHub in one form or another, I thought it might be a good idea to do the same with our main landing page. To that end, I'm trialling a [conversion of the website](https://github.com/red-eclipse/red-eclipse.github.io) to [GitHub Pages](https://pages.github.com/), you can see the result at [beta.redeclipse.net](http://beta.redeclipse.net/).

Currently, I'm just serving a static HTML page which I created using the base PHP script as a starting point (GH Pages only serves static content). My hope is we can convert the theme to a [Jekyll](https://help.github.com/articles/using-jekyll-as-a-static-site-generator-with-github-pages/) layout and start making more website content that the community at large has access to, possibly extending this to include a Blog, etc. in the near future.

Eventually, the main "redeclipse.net" HTTP won't serve any content and will just serve as a redirection point to each service. I'll also probably set up individual sub-domains for each section, for example: "www", "wiki", "blog" + .redeclipse.net.

**Comment by faergeek** [comment]
Hello. Is there anything you need help with? I'm a web developer, mostly frontend, so it's ideal way for me to contribute. The only thing is, I'm not designer at all.

Really like this game. Great work!!! 👍

**Comment by qreeves** [comment]
@faergeek Not sure about web frontend stuff, there's usually very little work unless we're doing a makeover or switch to another piece of software. As you can see, we're using Jekyll now, which only serves static pages.

I haven't finished the [GitHub issues API integration](https://redeclipse.net/discuss), and I was thinking of just keeping it simple by removing the login stuff and directing to the actual issues when trying to reply/etc. You could add the ability to reply/edit/etc if you really wanted to.

A lot of the old wiki needs porting to the new documents on the website too. It's hard to know what you're looking to do based in the information you've provided so far.

**Comment by faergeek** [comment]
About GitHub integration, I think it should be made better, at least searching is a must. But IMHO even better would be just linking directly to github issues, so there's no need for additional maintenance. And if you need to guide people who open issues you can use github issue templates. Take a look at [announcement blog post](https://blog.github.com/2016-02-17-issue-and-pull-request-templates/). There are also some [premade templates](https://github.com/stevemao/github-issue-templates).

Not sure about more interactive frontend stuff, I'm pretty sure I did have some ideas, can't remember right now. But anyway that may have negative cost, because at least I can't guarantee that I will maintain that stuff even a year from now. So I doubt it's worth it.

The only interactive thing I can think of right now is account creation. Not sure about how this works right now, but the last time there was some google forms stuff and you emailing me authkey :-) Not sure if for some reason it's feature, not a bug.

About migrating wiki, sign me in, that's kind of stuff that I have strange love to :-)

Is there any coordination about wiki migration? Sorry if I'm asking stupid question, didn't RTFM too much yet.

**Comment by faergeek** [comment]
So is there any help needed with the website? Or where can I read about wiki migration?

**Comment by IceflowRE** [comment]
#### Website
https://github.com/red-eclipse/red-eclipse.github.io
#### Wiki
https://github.com/red-eclipse/docs
https://github.com/red-eclipse/world/issues/9

**Comment by faergeek** [comment]
Thanks! 👍
